packages/xrpl/src/sugar/autofill.ts

```diff
@@ -140,41 +141,46 @@ export function setValidAddresses(tx: Transaction): void {
  * @param tagField - The field name for the tag in the transaction object.
  * @throws {ValidationError} If the tag field does not match the tag of the account address.
  */
-function validateAccountAddress(
-  tx: Transaction,
-  accountField: string,
-  tagField: string,
-): void {
+function validateAccountAddress<
+  T extends Transaction,
+  K extends keyof T & string,
+  K2 extends keyof T & string,
+>(tx: T, accountField: K, tagField: K2): void {
+  const val = tx[accountField]
+
+  if (typeof val !== 'string') {
+    throw new Error(`${accountField} must be a string`)
+  }
   // if X-address is given, convert it to classic address
-  const { classicAccount, tag } = getClassicAccountAndTag(tx[accountField])
-  // eslint-disable-next-line no-param-reassign -- param reassign is safe
-  tx[accountField] = classicAccount
+  const { classicAccount, tag } = getClassicAccountAndTag(val)
+  // eslint-disable-next-line no-param-reassign, @typescript-eslint/consistent-type-assertions -- param reassign is safe
+  ;(tx as Record<K, string>)[accountField] = classicAccount
 
   if (tag != null && tag !== false) {
-    if (tx[tagField] && tx[tagField] !== tag) {
+    if (tagField in tx && tx[tagField] !== tag) {
       throw new ValidationError(
         `The ${tagField}, if present, must match the tag of the ${accountField} X-address`,
       )
     }
-    // eslint-disable-next-line no-param-reassign -- param reassign is safe
-    tx[tagField] = tag
+    // eslint-disable-next-line no-param-reassign, @typescript-eslint/consistent-type-assertions -- param reassign is safe
+    ;(tx as Record<K2, number | false>)[tagField] = tag
```

Comment:
I'm not familiar with this syntax. Why do you have a statement terminator `;` at the beginning of this line?

Reply:
I don't really understand it either, but it's what's expected.